test/suite/test_hs12.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-2020 MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger, wttest, time
+
+def timestamp_str(t):
+    return '%x' % t
+
+# test_hs12.py
+# Verify we can correctly append modifies to the end of string values
+class test_hs12(wttest.WiredTigerTestCase):
+    conn_config = 'cache_size=2MB,statistics=(all),eviction=(threads_max=1)'
+    session_config = 'isolation=snapshot'
+
+    def test_modify_append_to_string(self):
```

Comment:
Could we test additional scenarios here? Inserting a modify at the beginning of a length 1 and of length > 1. I'm just looking to increase our test case coverage.

Reply:
@luke-pearson You mean the length of base value is 1 or the length of modify is 1?

Reply:
Just a modify of greater than one lenght.

Reply:
I added a test case but I'm not completely sure if it is what you are looking for. Let me know if you think more test cases are needed.

Reply:
Just set the 'AB' modify to be at position 0. So its at the front of the string, otherwise it looks good.